pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -523,6 +524,56 @@ def reset(self):
         self._opt_idx: Union[int, None] = None
         self._batch_size: Union[int, None] = None
         self._has_batch_loop_finished = False
+        self.legacy_batch_log_metrics = {}
+        self.legacy_batch_pbar_metrics = {}
+
+    def __call__(self,
+                 fx_name: Union[str, int, None] = None,
+                 dl_idx: Union[str, int, None] = None,
+                 opt_idx: Union[str, int, None] = None,
+                 batch_idx: Union[str, int, None] = None,
+                 split_idx: Union[str, int, None] = None,
+                 reduced=False):
+        """
+        This function is used to easily acces saved logged data.
+        """
+
+        hook_result = self[str(fx_name)]
+
+        dl_idx = str(dl_idx) if dl_idx is not None else None
+        opt_idx = str(opt_idx) if opt_idx is not None else None
+        batch_idx = str(batch_idx) if batch_idx is not None else None
+        split_idx = int(split_idx) if split_idx is not None else None
+
+        internal_type = hook_result._internal_type
+        if internal_type is None:
+            return Result()
+
+        if reduced:
+            result = hook_result._internals_reduced
+        else:
+            result = hook_result._internals
+
+        if internal_type == ResultStoreType.INSIDE_BATCH_TRAIN_LOOP:
+            if not reduced:
+                if dl_idx is not None:
```

Comment:
lets describe this on docs, that is is a chain 
`dl_idx -> opt_idx -> batch_idx -> split_idx`
and if any is missing it ends there...